internal/subscriber/retry/builder_test.go

```diff
@@ -0,0 +1,73 @@
+package retry
+
+import (
+	"sync"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/razorpay/metro/internal/brokerstore/mocks"
+	"github.com/razorpay/metro/internal/credentials"
+	"github.com/razorpay/metro/internal/subscription"
+	mocks2 "github.com/razorpay/metro/pkg/cache/mocks"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRetrier_Build(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	bs := mocks.NewMockIBrokerStore(ctrl)
+	ch := mocks2.NewMockICache(ctrl)
+	subs := getMockSubscriptionModel()
+	errCh := make(chan error)
+
+	expectedRetrier := &Retrier{
+		subscriberID:   "subscription-id",
+		subs:           subs,
+		bs:             bs,
+		ch:             ch,
+		backoff:        NewFixedWindowBackoff(),
+		finder:         NewClosestIntervalWithCeil(),
+		handler:        NewPushToPrimaryRetryTopicHandler(bs),
+		delayConsumers: sync.Map{},
+		errChan:        errCh,
+	}
+
+	retrierBuilder := NewRetrierBuilder().
+		WithSubscriberID("subscription-id").
+		WithBrokerStore(bs).
+		WithSubscription(subs).
+		WithBackoff(NewFixedWindowBackoff()).
+		WithCache(ch).
+		WithErrChan(errCh).
+		WithIntervalFinder(NewClosestIntervalWithCeil()).
+		WithMessageHandler(NewPushToPrimaryRetryTopicHandler(bs))
+	assert.Equalf(t, expectedRetrier, retrierBuilder.Build(), "Build()")
```

Comment:
why are we using equalf and not equal..?

Reply:
I thought EqualF will print detailed difference if Equalf assertion failed. since all the variable are mocked it is not required here. removing Equalf